### PR TITLE
migrate normalizeAPIError to TS

### PR DESCRIPTION
**Pull request by Feranchz**
Relates to https://github.com/strapi/strapi/issues/17690

### CODE REVIEW

**Review by markkaylor — APPROVED** [review]
LGTM

**Review by joshuaellis — CHANGES_REQUESTED** [review]
After some research the API error we're passing is one of the errors in `@strapi/utils`, they're classes but right now we can't access them. We can either delay this PR and probably #17922 until we've sorted `@strapi/utils` out (a couple of days max i imagine) or we can do a `namespace` to force the types. 

What do you think @markkaylor & @Feranchz?

**Comment by joshuaellis** [comment]
I can add the namespace work to this PR @markkaylor if you want?

File: packages/core/utils/src/index.ts [hunk 1]
```diff
@@ -43,7 +46,58 @@ import * as traverse from './traverse';
 import webhook from './webhook';
 import { isOperator, isOperatorOfType } from './operators';
 
-export = {
+// @ts-expect-error duplicate because we're namespacing to patch around the fact we don't have types exporting correctly atm.
+namespace utils {
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/core/utils/src/index.ts:50]: yep, don't worry i'm still looking at it 😄

File: packages/core/helper-plugin/declarations/@strapi/utils.d.ts
Comment (joshuaellis) [inline packages/core/helper-plugin/declarations/@strapi/utils.d.ts:1]: Right, this is my workaround at the moment. I tried to do what @alexandrebodin is doing on his `@strapi/strapi` branch at the moment but with classes it through the issue of two `utils` being defined at the same time. I tried _numerous_ times to get the `custom.d.ts` to work as well but again, it couldn't patch over the broken types so after a lot of trial and error and googling, this came up as the best solution for now...

This is something we're gonna have to do with `@strapi/permission` too and then implement the same way in other packages until those offending packages are bundled correctly, what does everyone else think – @markkaylor @Feranchz @gu-stav?

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**